src/core/public/overlays/sidecar/components/sidecar.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import useObservable from 'react-use/lib/useObservable';
+import { BehaviorSubject } from 'rxjs';
+import classNames from 'classnames';
+import { I18nStart } from '../../../i18n';
+import { MountPoint } from '../../../types';
+import { MountWrapper } from '../../../utils';
+import './sidecar.scss';
+import { ResizableButton } from './resizable_button';
+import { ISidecarConfig, OverlaySidecarOpenOptions, SIDECAR_DOCKED_MODE } from '../sidecar_service';
+
+export interface Props {
+  sidecarConfig$: BehaviorSubject<ISidecarConfig | undefined>;
+  options: OverlaySidecarOpenOptions;
+  setSidecarConfig: (config: Partial<ISidecarConfig>) => void;
+  i18n: I18nStart;
+  mount: MountPoint;
+}
+
+export const Sidecar = ({ sidecarConfig$, options, setSidecarConfig, i18n, mount }: Props) => {
+  const sidecarConfig = useObservable(sidecarConfig$, undefined);
+
+  const classes = classNames(
+    'osdSidecarFlyout',
+    {
+      'osdSidecarFlyout--dockedRight': sidecarConfig?.dockedMode === SIDECAR_DOCKED_MODE.RIGHT,
+      'osdSidecarFlyout--dockedLeft': sidecarConfig?.dockedMode === SIDECAR_DOCKED_MODE.LEFT,
+      'osdSidecarFlyout--dockedTakeover':
+        sidecarConfig?.dockedMode === SIDECAR_DOCKED_MODE.TAKEOVER,
+      'osdSidecarFlyout--hide': sidecarConfig?.isHidden === true,
+    },
+    options.className
+  );
+  const handleResize = useCallback(
+    (newSize: number) => {
+      setSidecarConfig({
+        paddingSize: newSize,
+      });
+    },
+    [setSidecarConfig]
+  );
+
+  const flyoutSizeStyle = useMemo(
+    () =>
+      sidecarConfig?.dockedMode === SIDECAR_DOCKED_MODE.TAKEOVER
+        ? {
+            height: sidecarConfig?.paddingSize,
+          }
+        : {
+            width: sidecarConfig?.paddingSize,
+          },
+    [sidecarConfig]
+  );
+
+  return (
+    <i18n.Context>
+      <div data-test-subj={options['data-test-subj']} style={flyoutSizeStyle} className={classes}>
+        <ResizableButton
+          onResize={handleResize}
+          dockedMode={sidecarConfig?.dockedMode}
+          flyoutSize={sidecarConfig?.paddingSize ?? 0}
```

Comment:
I can understand you mean. I try to do some updates, but i find it's hard to decouple. 

`resizableButton` needs to get `dockedMode` to display vertical or horizontal style.  `onMouseMove` and `onMouseDown` both need to get `dockedMode` and `flyoutSize` to calculate new `paddingSize`, rather than only one event. So if we remove `dockedMode` and `flyoutSize`, we need to add other more parameters, like `isHorizontal`, `onMouseMove`,`onMouseDown`, i'm not sure if this will make it more confusing. 

I think there is a way to reduce parameters is to render buttons and containers in one component. I do some research on [similar component in other library](https://github.com/elastic/eui/blob/main/src-docs/src/views/resizable_container/resizable_button.tsx#L69), I find although it's named as `resizableButton`, it actually render both button and container.  If here we update to render container and button together, `resizableButton` will not be pure. So I think it could be difficult to update here.